ark/mibi/tiling_utils.py

```diff
@@ -446,22 +474,672 @@ def create_tiled_regions(tiling_params, moly_point, tma=False):
             fov = copy.deepcopy(tiling_params['fovs'][region_index])
             fov['centerPointMicrons']['x'] = cur_x
             fov['centerPointMicrons']['y'] = cur_y
-            fov['name'] = f'row{yi}_col{xi}'
+            fov['name'] = fov_names[index]
 
-            # append value to tiled_regions
-            tiled_regions['fovs'].append(fov)
+            # append value to fov_regions
+            fov_regions['fovs'].append(fov)
 
             # increment moly_counter as we've added another fov
             moly_counter += 1
 
             # append a moly point if moly_interval is set and we've reached the interval threshold
             if 'moly_interval' in tiling_params and \
                moly_counter % tiling_params['moly_interval'] == 0:
-                tiled_regions['fovs'].append(moly_point)
+                fov_regions['fovs'].append(moly_point)
 
         # append moly point to seperate runs if not last and if specified
         if tiling_params['moly_run'] == 'Y' and \
            region_index != len(tiling_params['region_params']) - 1:
-            tiled_regions['fovs'].append(moly_point)
+            fov_regions['fovs'].append(moly_point)
+
+    return fov_regions
+
+
+def convert_microns_to_pixels(coord):
+    """Convert the coordinate in stage microns to optical pixels
+
+    In other words, co-register using the centroid of a FOV.
+
+    The values are coerced to ints to allow indexing into the slide.
+    Coordinates are also returned in (y, x) form to account for a different coordinate axis.
+
+    Args:
+        coord (tuple):
+            The coordinate in microns to convert
+
+    Returns:
+        tuple:
+            The converted coordinate from microns to pixels
+    """
+
+    # NOTE: all conversions are done using the fiducials
+    # convert from microns to stage coordinates
+    stage_coord_x = (coord[0] * 0.001001 - 0.3116)
+    stage_coord_y = (coord[1] * 0.001018 - 0.6294)
+
+    # convert from stage coordinates to pixels
+    pixel_coord_x = (stage_coord_x + 27.79) / 0.06887
+    pixel_coord_y = (stage_coord_y - 77.40) / -0.06926
+
+    return (int(pixel_coord_y), int(pixel_coord_x))
+
+
+def assign_closest_fovs(manual_fovs, auto_fovs, moly_point_name):
```

Comment:
This function can be simplified now that we're only relabeling existing TMAs. Instead of taking in a fully-formed JSON, I think it can just take in the names and centroids of the automatically generated FOVs. 

We can assume that there aren't any moly points in the manually generated TMA, and remove that logic from here

Reply:
Yeah I agree, don't need to pass in the whole FOV list for either `manual_fovs` and `auto_fovs`. Nuking `set_tiling_params` and having a dedicated FOV JSON creator for TMAs will remove the need for Moly point checking here.

Reply:
Alternatively, since `assign_closest_fovs` currently creates `manual_fovs_info` and `auto_fovs_info` which is just a `dict` mapping from each FOV to its centroid, I could just have the notebook do that and pass those in directly as arguments. Would this work or is there a reason we just want separate lists of names and centroids for both manual and auto FOVs?

Reply:
I think it's okay to leave the dict creation inside of the function, rather than have it in the notebook, to keep things clean. 

My comment was more related to what the automatic FOV logic needs to be. We don't need to create an entire fake JSON containing the automatically generated FOVs, since that JSON will never be saved anywhere. The only thing we need to do is create the centroids in the right spot with the right names, so that they can be used to relabel the manual JSON.